src/asmcnc/core_UI/ScreenDesigner/designer_controller.py

```diff
@@ -0,0 +1,190 @@
+        self.widget_to_add_to = None
+
+        # create designer screen
+        self.screen_manager = App.get_running_app().sm
+        self.component_widget = None
+        self.screen_drop_down = None
+        self.screen_name_input = None
+
+        self.designer_screen = None
+
+        self.available_screens = {}
+        self.modifying_screen = False
+        self.current_screen_name = ''
+        self.current_screen_layout = None
+
+    def load_generated_screens(self, *args):
+        """
+        Walks through all files in ScreenDesigner/generated_screens and populates the dropdown with the
+        screens names.
+        Saves a dict with the screen name and its import path for later use.
+        Assumes that screen name == class name. The import path ("asmcnc.core_UI.screen_designer")
+        derives from the file path.
+        """
+        children = [c for c in self.screen_drop_down.children[0].children]
+        for c in children:
+            self.screen_drop_down.remove_widget(c)
+        path = pu.get_path('generated_screens')
+        for screen_file in os.listdir(path):
+            prepped_filename = screen_file.replace('.py', '')
+            class_name = self.screenname_to_filename(prepped_filename, True)
+            import_path = path.replace('/', '.').split('src')[1][1:] + '.' + prepped_filename
+            if screen_file.startswith('__init__') or screen_file.endswith('.pyc'):
+                continue
+            # fill dropdown with class names
+            btn = ButtonBase(text=class_name, size_hint_y=None, height=40, id='DESIGNER')
+            btn.bind(on_release=lambda btn: self.screen_drop_down.select(btn.text))
+            self.screen_drop_down.add_widget(btn)
+            # saves class names (=screen names) and import path for later
+            self.available_screens[class_name] = import_path
+
+
+    @staticmethod
+    def screenname_to_filename(name_to_convert, reverse=False):
+        # type: (str, bool) -> str
+        """
+        reverse = False:
+            takes a screen name (e.g. MyFirstScreen) and returns the filename (my_first_screen)
+
+        reverse = True:
+            takes a filename (e.g. my_first_screen) and return a screen name (MyFirstScreen)
+        """
+        if reverse:
+            return re.sub(r'_([a-z])', lambda pat: pat.group(1).upper(), '_' + name_to_convert)
+        else:
+            return re.sub(r'([A-Z])', lambda pat: '_' + pat.group(1).lower(), name_to_convert)[1:]
+
+    def create_screen(self, screen_name):
+        """
+        Creates a new instance of the selected screen. Imports the corresponding module first.
+        """
+        try:
+            test = __import__(name=self.available_screens[screen_name], fromlist=[screen_name])
+            reload(test)
+            screen_class = getattr(test, screen_name)
+            screen = screen_class()
+            return screen
+        except ImportError as ex:
+            Logger.exception(ex)
```

Comment:
I think Logger.exception should already output the exception traceback, might be worth instead having a hint message inside.